pycalphad/models/model_mqmqa.py

```diff
@@ -276,6 +281,18 @@ def _X_i(self, dbe, species: v.Species):
             assert species in self.anions
             return self._n_i(dbe, species) / sum(self._n_i(dbe, x) for x in self.anions)
 
+    def _X_i_S(self, dbe, species: v.Species):
+        """
+        Return the site fraction of species on it's sublattice. Poschmann Eq. 9 and 10.
+        """
+        if species in self.cations:
+            cons_spec=[cons for cons in species.constituents.values()][0]
+            return (self._n_i(dbe, species)/cons_spec) / sum((self._n_i(dbe, a)/cons) for a in self.cations for cons in a.constituents.values())
+        else:
+            assert species in self.anions
+            cons_spec=[cons for cons in species.constituents.values()][0]
+            return (self._n_i(dbe, species)/cons_spec) / sum((self._n_i(dbe, x)/cons) for x in self.anions for cons in x.constituents.values())
+
```

Comment:
Does your implementation of `_X_i_S` here simply replace `_X_i` (i.e. current `_X_i` is wrong)? There's now no place that `_X_i` is used in the code.

If so, we should just fix `_X_i` with your change instead of making a new method.

Reply:
Okay. I made the change.